rfcs/0140-simple-package-paths.md

```diff
@@ -0,0 +1,363 @@
+- (-) Enforcing this requires a global view of Nixpkgs, which is nasty to implement
+- (-) [Package variants](#package-variants) would not be possible to define
+
+## Allow `callPackage` arguments to be specified in `<unit>/args.nix`
+
+Context: Custom `callPackage` arguments have to be added to `all-packages.nix`
+
+Alternative:
+Expand the auto-calling logic according to:
+Unit directories are automatically discovered and transformed to a definition of the form
+```
+# If args.nix doesn't exist
+pkgs.${name} = pkgs.callPackage ${unitDir}/package.nix {}
+# If args.nix does exists
+pkgs.${name} = pkgs.callPackage ${unitDir}/package.nix (import ${unitDir}/args.nix pkgs);
+```
+
+- (+) It makes another class of packages uniform, by picking a solution with restricted expressive power.
+- (-) It does not solve the contributor experience problem of having too many rules.
+      `args.nix` is another pattern that contributors need to learn how to use, as we have seen that it is not immediately obvious to everyone how it works.
+  - (+) A CI check can mitigate the possible lack of uniformity, and we see a simple implementation strategy for it.
+- (-) Complicates the unit directory structure with an optional file
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+# Future work
+[future]: #future-work
+
+All of these questions are in scope to be addressed in future discussions in the [Nixpkgs Architecture Team](https://nixos.org/community/teams/nixpkgs-architecture.html):
+
+- Expose an API to get access to the package functions directly, without calling them
+- Add a meta tagging or categorization system to packages as a replacement for the package categories. Maybe `meta.tags` with `search.nixos.org` integration. Maybe https://repology.org/ integration. See also https://github.com/NixOS/rfcs/pull/146.
+- Making the filetree more human-friendly by grouping files together by "topic" rather than technical delineations.
+  For instance, having a package definition, changelog, package-specific config generator and perhaps even NixOS module in one directory makes work on the package in a broad sense easier.
+- This RFC only addresses the top-level attribute namespace, aka packages in `pkgs.<name>`, it doesn't do anything about package sets like `pkgs.python3Packages.<name>`, `pkgs.haskell.packages.ghc942.<name>`, which may or may not also benefit from a similar auto-calling
+- Improve the semantics of `callPackage` and/or apply a better solution, such as a module-like solution
+- Potentially establish an updateScript standard to avoid problems like, relates to Flakes too
+- What to do with different versions, e.g. `wlroots = wlroots_0_14`? This goes into version resolution, a different problem to fix
+- What to do about e.g. `libsForQt5.callPackage`? This goes into overrides, a different problem to fix
+- What about aliases like `jami-daemon = jami.jami-daemon`?
```

Comment:
There's an answer missing here ;) I believe it might be good enough to keep them in `all-packages.nix`, just as with custom args,  for now, in order to keep the scope small?

Reply:
yeah seems fair